### PR TITLE
Support frontend hot reload on Linux

### PR DESCRIPTION
**Pull request by gok99**
**Proposed commit message:**

```
The hotReloadFrontend gradle task does not support Linux. Apache
ant's `Os` class does not have an OS family specifically for Linux.
The Mac hot reload feature also does not appear to work.

Let's:
- add hot reload for Linux using gradle's OperatingSystem::isLinux
- use `npm run serveOpen` command for both Mac and Linux
```

Tested on Windows, Mac and Fedora Linux

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -228,22 +230,30 @@ task syncFrontendPublic(type: Sync) {
...
+    onlyIf {os.isWindows()}
     workingDir 'frontend/'
     commandLine 'cmd','/c', 'START', '"hotreload RepoSense frontend"', 'npm', 'run', 'serveOpen'
 }
 
+task linuxHotReloadFrontend(type: Exec) {
+    dependsOn(installFrontend)
+    onlyIf {os.isLinux()}
+    workingDir 'frontend/'
+    commandLine 'npm', 'run', 'serveOpen'
```
Comment (gok99) [inline build.gradle:249]: This works for me, but I'm not sure if there's a canonically correct way to write this.

**Comment by gerhean** [comment]
For Windows OS, the reason why I spawn a new window for hotreload is due to Gradle causing the process to be orphaned (I think).

In other words, if I `npm run serveOpen` without openning new window, doing  `ctrl-c` to kill the application  will cause the Vue process to run in the background and not terminate properly. 

Can you check if such is also the case for linux?

File: build.gradle (hunk 1 above)
Comment (gerhean) [reply build.gradle:249]: There is no canonically correct way, there is only consistency, and I find it consistent other than the fact that this one doesn't spawn the `serveOpen` in a new window which may or may not be needed.

**Review by gerhean — APPROVED** [review]
LGTM. No regressions on Windows.

**Review by fzdy1914 — APPROVED** [review]

**Comment by gok99** [comment]
> Can you check if such is also the case for linux?

The `vue-cli-service serve` process seems to stop after a `ctrl-c` and the locally hosted site is no longer accessible.

**Review by fzdy1914 — CHANGES_REQUESTED** [review]
See our discussion in slack for more information.

**Review by fzdy1914 — APPROVED** [review]

**Comment by dcshzj** [comment]
Confirmed to work on ARM-based Macs as well.

**Pull request merged**